### ISSUE TITLE
Mercado pago does not work when attempting to pay

### ISSUE THREAD

**Issue by Paul-Eraman-CoopCycle** [coopcycle/coopcycle-web#3701]
**Describe the bug**
when trying to pay with Mercado pago you receive the "Bad_request: deferred capture not supported" message

**To Reproduce**
1. Go to pay for a food delivery order on the bogata instance
2. Attempt to pay with mercado pago
3. See error

**Expected behavior**

**Screenshots**
![imagen](https://github.com/coopcycle/coopcycle-web/assets/77277854/b7a94716-45cc-474b-9ede-37582d5eb670)

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by lucasferraro** [comment]
The error "Deferred capture not supported" is related to the capture of a payment after an order is completed.

I didn't know this but MP does not support "deferred capture" for all payment methods and this depends in for which country too. For example for Colombia only Visa Credit Card supports this feature. 
I guess that this error happens because they are using a Card different to Visa Credit Card.

So, we should change the implementation of MP to handle this in a better way:

- [ ] Check if payment method (PM) used by the user allows deferred capture.
- [ ] If current PM doesn't support deferred capture we would have to capture the payment at the same time that we create in MP.
- [ ] Also, if current PM doesn't support deferred capture, we shouldn't perform the capture operation to MP after an order is completed.
- [ ] If current PM supports deferred capture the existing flow should work without problems.

Finally, during this research I figured out that now MP is forcing to update the JavaScript SDK that we are using in the front end. [Here](https://www.mercadopago.com.ar/developers/en/docs/sdks-library/client-side/java/howto-migrate) there's a guide to perform that migration. For now the migration is only required when we use TEST credentials, so in production this should not affect so far, but I suggest to move forward with this migration and avoid unexpected problems in a near feature in production.

- [ ] Migrate MP SDK JS V1 to SDK JS V2 with Secure Fields

**Comment by lucasferraro** [comment]
- [x] MP SDK migrated to V2

New front end implementation for MercadoPago using the new concept of Bricks
https://www.mercadopago.com.ar/developers/en/docs/checkout-bricks/card-payment-brick/introduction


https://github.com/coopcycle/coopcycle-web/assets/4819244/80511bdb-cf9a-4413-84cc-e23a5dd2d303

**Comment by Paul-Eraman-CoopCycle** [comment]
Do we need to test this? or good to go?

**Comment by lucasferraro** [comment]
> Do we need to test this? or good to go?

Yes we need a test and feedback from Donky Bogotá, because the error is related to the "deferred capture" feature that in Colombia is only supported for Visa Credit Card. So they should test if it works now with Visa Credit and another Credit card as well.

**Comment by Paul-Eraman-CoopCycle** [comment]
Based on the crazyness of this subject today, @alexsegura do we need to wait on testing this, or can i tell them to start testing it now? Monday?
